ts/src/pro/blofin.ts

```diff
@@ -0,0 +1,489 @@
+                'watchOrderBook': {
+                    'channel': 'books',
+                },
+                'watchOrderBookForSymbols': {
+                    'channel': 'books',
+                },
+            },
+            'streaming': {
+                'ping': this.ping,
+                'keepAlive': 25000, // 30 seconds max
+            },
+        });
+    }
+
+    ping (client) {
+        return 'ping';
+    }
+
+    handleWsPong (client: Client, message) {
+        //
+        //   'pong'
+        //
+        client.lastPong = this.milliseconds ();
+    }
+
+    async watchTrades (symbol: string, since: Int = undefined, limit: Int = undefined, params = {}): Promise<Trade[]> {
+        /**
+         * @method
+         * @name blofin#watchTrades
+         * @see https://docs.blofin.com/index.html#ws-trades-channel
+         * @description get the list of most recent trades for a particular symbol
+         * @param {string} symbol unified symbol of the market to fetch trades for
+         * @param {int} [since] timestamp in ms of the earliest trade to fetch
+         * @param {int} [limit] the maximum amount of trades to fetch
+         * @param {object} [params] extra parameters specific to the exchange API endpoint
+         * @returns {object[]} a list of [trade structures]{@link https://docs.ccxt.com/#/?id=public-trades}
+         */
+        params['callerMethodName'] = 'watchTrades';
+        return await this.watchTradesForSymbols ([ symbol ], since, limit, params);
+    }
+
+    async watchTradesForSymbols (symbols: string[], since: Int = undefined, limit: Int = undefined, params = {}): Promise<Trade[]> {
+        /**
+         * @method
+         * @name blofin#watchTradesForSymbols
+         * @see https://docs.blofin.com/index.html#ws-trades-channel
+         * @description get the list of most recent trades for a list of symbols
+         * @param {string[]} symbols unified symbol of the market to fetch trades for
+         * @param {int} [since] timestamp in ms of the earliest trade to fetch
+         * @param {int} [limit] the maximum amount of trades to fetch
+         * @param {object} [params] extra parameters specific to the exchange API endpoint
+         * @returns {object[]} a list of [trade structures]{@link https://docs.ccxt.com/#/?id=public-trades}
+         */
+        await this.loadMarkets ();
+        const trades = await this.watchMultipleWrapper ('trades', 'watchTradesForSymbols', symbols, params);
+        if (this.newUpdates) {
+            const first = this.safeDict (trades, 0);
+            const tradeSymbol = this.safeString (first, 'symbol');
+            limit = trades.getLimit (tradeSymbol, limit);
+        }
+        return this.filterBySinceLimit (trades, since, limit, 'timestamp', true);
+    }
+
+    handleWsTrades (client: Client, message) {
```

Comment:
@ttodua can you restore the handleX methods names? we don't have this method in REST and all other exchanges call it handleTrades/handleorderBook/etc, there's o reason for changing them here

Reply:
done.